lib/chartmogul/subscription.rb

```diff
@@ -14,9 +14,12 @@ class Subscription < APIResource
     include API::Actions::Custom
 
     def set_cancellation_dates(cancellation_dates_array)
-      @cancellation_dates = cancellation_dates_array.map do |cancellation_date|
-        Time.parse(cancellation_date)
-      end
+      @cancellation_dates = parse_dates(cancellation_dates_array)
```

Comment:
I would argue it was supposed to make the calls in the first place and not doing them is unexpected behavior which should be fixed.
We document it as such: https://dev.chartmogul.com/reference#cancel-a-customers-subscription
`subscription.set_cancellation_dates([Time.utc(2016, 1, 1), Time.utc(2017, 1, 1)])`

Therefore I would just make this an alias to `update_cancellation_dates`, otherwise it will be highly confusing (and not doing anything silently - the worst kind of errors)

Reply:
Ok, it looked to me that the setter was overridden in order to work here: https://github.com/chartmogul/chartmogul-ruby/blob/master/lib/chartmogul/object.rb#L87

Also, whenever we will be setting the attribute it will be making a call. So for example, if we set a different attribute of a subscription, we will make a call to update the subscription. When we receive the updated subscription in the response we will set the attributes of the subscription and we will send an API request to update the cancellation dates. Thus we have an infinite loop.

So I think we cannot make the setter call the API. WDYT?

Reply:
I wasn't aware of this metaprogramming... you're correct. But we need to ensure ppl don't call it thinking it makes a request. Can we make the `set_` method private?

Reply:
Sure, I will try that.

Reply:
Yeah, it isn't. But at least it will raise an error, so people fix their code, which is my intention. As it never really worked as advertised, it's only proper.